### PR TITLE
Update electron to version 30.1.2

### LINKED ISSUE #14014: Update Electron Version
### Feature Description:
Due to https://github.com/eclipse-theia/theia/issues/14011 it's a good time for another update of our electron dependency to a suitable newer version. VS Code is currently at 30.1.2.

### PR DESCRIPTION
**Pull request by tsmaeder**
#### What it does
Fixes #14014

Contributed on behalf of STMicroelectronics

#### How to test
Make sure everything builds and test general functionality in the electron example. Make sure there are no new error messages in the log

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by tsmaeder** [comment]
Draft PR while we wait for IP-Ticket approval

**Review by planger — APPROVED** [review]
I tested
* opening Electron Theia
* Keyboard shortcuts
* Application menus
* Context menu
* Opening a new instance via Welcome screen
* Native dialogs via Save File dialog and Open Folder dialog

I noticed two issues (see below), however, I could reproduce both with `master`, so they are not caused by the Electron update.

## Unrelated observed issues

Secondary window didn't close:
1. Extract text editor into secondary window
2. Make changes
3. Close window and select "Don't save"
4. Window stays open and is empty, cannot be closed

In the console I get the following error:

```text
(node:133669) UnhandledPromiseRejectionWarning: Error: Cancelled
    at cancelled (/home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:427:12)
    at /home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:5570:54
    at /home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1410:69
    at CallbackList.invoke (/home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1416:26)
    at Emitter.fire (/home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:1531:36)
    at MutableToken.cancel (/home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:362:31)
    at CancellationTokenSource.cancel (/home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/packages_core_lib_common_index_js-node_modules_vscode-languageserver-types_lib_umd_sync_recursive.js:407:25)
    at showWindowAndCloseSplashScreen (/home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/electron-main.js:1393:31)
    at /home/philip/Git/OpenSource/Theia/theia/examples/electron/lib/backend/electron-main.js:1401:36
(Use `electron --trace-warnings ...` to show where the warning was created)
(node:133669) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
```

**Pull request merged**
